helix-view/src/modeline.rs

```diff
@@ -0,0 +1,258 @@
+use std::borrow::Cow;
+
+use helix_core::indent::IndentStyle;
+use helix_core::regex::Regex;
+use helix_core::{LineEnding, Rope};
+use once_cell::sync::Lazy;
+
+// 5 is the vim default
+const LINES_TO_CHECK: usize = 5;
+const LENGTH_TO_CHECK: usize = 256;
+
+static MODELINE_REGEX: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"(?:\S+\s+)?(?:vi|vim|Vim|ex):\s*(?:set? )?(.*)(?::[^:]*)?").unwrap());
```

Comment:
The regec should be anchored (that makes it both faster and correct since the modeline is specific about how it may be prefixed We also don't need capture groups here. Since modeline support escapes (with `\:` we can't phrase the option with regec anyway (it's technically possible but not worth the effort. You can simply find the regec match and then use the rest of the line (from the start if the regec). That also removes the need for non-capturing groups (since you will not compute captures anywya which sinplfies the regec a bit further

```suggestion
    Lazy::new(|| Regex::new(r"^(\S*\s+)?(vi|[vV]im[<=>]?\d*|ex):\s*(set?\s+)?").unwrap());
```